### PR TITLE
Setup public area - pre CB login

### PR DESCRIPTION
**Pull request by RogeredBacon**
Added various buttons and password reset page with email functionality; though no actual password reset yet. Changed the schema to add time-stamps, unique constraint for cbusiness email, genre varchar size increase.

### CODE REVIEW

**Comment by ZooeyMiller** [comment]
since this contains all of the changes from #93 I'm going to wait until that PR is merged to review this one, so it's easier to read.

File: react-backend/routes/cbauthentication/checkCBemail.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          }
+        }
+      });
+    } else if (!validator.isEmail(data.formEmail)) {
+      console.log('This isnt a correct email!?');
+      res.send('email');
+    }
+  });
+});
+
```
Comment (ZooeyMiller) [inline react-backend/routes/cbauthentication/checkCBemail.js]: this file is called `checkCBemail.js`, but I don't feel like that sums it up. This file checks if the CB exists **and** sends a password reset email.

I feel like some kind of naming which pertains to the fact that this is the route which triggers the password email would be more relevant.

File: react-backend/routes/cbauthentication/checkCBemail.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const validator = require('validator');
+const getCBAlreadyExists = require('../../database/queries/getCBAlreadyExists');
+const express = require('express');
+const sendresetemail = require('../../functions/sendresetemail');
```
Comment (ZooeyMiller) [inline react-backend/routes/cbauthentication/checkCBemail.js]: camelcase plz :pray:

File: react-backend/app.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ const activities = require('./routes/activities');
 const checkCB = require('./routes/cbauthentication/checkCB');
 const registerCB = require('./routes/cbauthentication/registerCB');
 const checkCBlogin = require('./routes/cbauthentication/checkCBlogin');
+const checkCBemail = require('./routes/cbauthentication/checkCBemail');
```
Comment (ZooeyMiller) [inline react-backend/app.js]: This has now been changed to `CBPasswordResetInstigator.js` so needs to be changes here

File: react-backend/app.js [hunk 4]
```diff
@@ -42,6 +43,7 @@ app.use('/activities', activities);
 app.use('/checkCB', checkCB);
 app.use('/registerCB', registerCB);
 app.use('/checkCBlogin', checkCBlogin);
+app.use('/checkCBemail', checkCBemail);
```
Comment (ZooeyMiller) [inline react-backend/app.js]: also needs to be updated to `CBPasswordResetInstigator` here (or whatever you decide to call it when you require it in above).

File: react-backend/client/src/components/authentication/pswdresetcb.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    const checkData = {
+      formEmail: this.state.email
+    };
+
+    fetch('/checkCBemail', {
```
Comment (ZooeyMiller) [inline react-backend/client/src/components/authentication/pswdresetcb.js]: If you change the path in the router as noted above then you'll need to change it here too.

**Comment by RogeredBacon** [comment]
okie dokie I think I've updated the stuff

**Comment by ZooeyMiller** [comment]
@RogeredBacon Okay cool! I didn't see this at first, just remember to assign me and add an awaiting review label to ensure I see it.

**Pull request merged**